internal/measurer/measurer.go

```diff
@@ -0,0 +1,127 @@
+// Package measurer contains measurement code
+package measurer
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/ooni/probe-engine/internal/dialer"
+	"github.com/ooni/probe-engine/internal/httptransport"
+	"github.com/ooni/probe-engine/internal/resolver"
+	"github.com/ooni/probe-engine/internal/tlsdialer"
+)
+
+// Logger is the logger interface assumed by this package
+type Logger interface {
+	Debugf(format string, v ...interface{})
+}
+
+// Config contains the settings for the measurer.
+type Config struct {
+	Logger Logger
+}
+
+// Results contains the measurement results.
+type Results struct {
+	Connects       []dialer.Events
+	HTTPBodies     []httptransport.BodySnapshot
+	HTTPRoundTrips []httptransport.Events
+	Resolutions    []resolver.Events
+	TLSHandshakes  []tlsdialer.Events
+}
+
+// Get performs a GET request and returns the measurement results. The error
+// signals whether the request failed. You will get results in any case, since
+// the purpose of this function is to measure anomalies :^).
+func Get(url string, config Config) (Results, error) {
```

Comment:
This function shows that we can obtain the same measurements we're currently obtaining with the `MeasurementRoot` stored inside the context without using the context at all (at least at this level of abstraction).